### PR TITLE
[fix #1046] fix  cross-origin url cannot be redirected when  "externalLinkTarget" is set to "_self" and "routerMode" is set to "history".

### LINKED ISSUE #1046: A cross-origin url cannot be redirected to if "externalLinkTarget" is set to "_self" and "routerMode" is set to "history"
#### Steps to reproduce

Set the options firstly:

```javascript
window['$docsify'] = {
  routerMode: 'history',
  externalLinkTarget: '_self'
}
```

Then write a link whose url is from another domain:

```markdown
Example: [example.com](https://example.com/)
```

#### What is current behaviour

If our domain were `docsify.now.sh`, it would do nothing and pop an error on the debug console:

```
Uncaught DOMException: Failed to execute 'pushState' on 'History': A history state object with URL 'https://example.com/' cannot be created in a document with origin 'https://docsify.now.sh' and URL 'https://docsify.now.sh/'.
    at https://unpkg.com/docsify@4.10.2/lib/docsify.min.js:1:61459
```

The corresponding source code is [here](https://github.com/docsifyjs/docsify/blob/v4.10.2/src/core/router/history/html5.js#L30).

#### What is the expected behaviour

Redirect to `https://example.com/` normally.

#### Other relevant information

- [x] Bug does still occur when all/other plugins are disabled?

- Your OS: `Linux ZZZ-PC-4 5.4.0-4-amd64 #1 SMP Debian 5.4.19-1 (2020-02-13) x86_64 GNU/Linux`
- Node.js version: `v12.14.0`
- npm/yarn version: `6.13.4`
- Browser version: `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.122 Safari/537.36`
- Docsify version: `4.10.2`
- Docsify plugins: `None`

### PR DESCRIPTION
**Pull request by Koooooo-7**
FIX #1046 
Cross-origin url cannot be redirected when  "externalLinkTarget" is set to "_self" and "routerMode" is set to "history".

Now, When  "externalLinkTarget"  set to "_self" and "routerMode", we will use `windows.open`  open url in current page instead.
what you should do is adding the `:crossorigin` next to  ur url
eg:
```md
[example.com](https://example.com/ ':crossorgin')  
```

**Summary**

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature
- [x] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

---

* [x] DO NOT include files inside `lib` directory.

### CODE REVIEW

File: docs/helpers.md [hunk 1]
```diff
@@ -65,6 +65,12 @@ You will get `<a href="/demo/">link</a>`html. Do not worry, you can still set ti
 [link](/demo ':disabled')
 ```
 
+## Cross-Origin link
+Only when you set the `routerMode: 'history'` and `externalLinkTarget: '_self'` , you need add this configuration for those Cross-Origin links
```
Comment (anikethsaha) [inline docs/helpers.md:69]: need a newline here

**Comment by anikethsaha** [comment]
also, about the commit message format [refer this guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md)

File: docs/helpers.md (hunk 1 above)
Comment (anikethsaha) [reply docs/helpers.md:69]: after `L68`

Comment (Koooooo-7) [reply docs/helpers.md]: yep, I updated it.

**Pull request merged**